game.py

```diff
@@ -1,232 +1,265 @@
-            if placed == 3:
-                break
-            x = random.randint(150, 600)
-            y = random.randint(150, 600)
-            x, y = normalize_cords(x, y)
-            x_grid = int((x / 50) - 3)
-            y_grid = int((y / 50) - 3)
-            if self.space[x_grid][y_grid] != 0:
-                continue
-            color = next_colors.pop()
-            self.sqr_grid[x_grid][y_grid] = ColoredRect(
-                color,
-                x,
-                y
-            ).draw_colored_rect(color)
-            self.space[x_grid][y_grid] = 1
-            placed += 1
-
-
-if __name__ == '__main__':
-    AlignIt()
+from __future__ import annotations
+
+import random
+import sys
+from time import sleep
+
+import pygame
+
+from astar import astar
+from coloredRect import ColoredRect
+
+
+WINDOW_HEIGHT = 600
+WINDOW_WIDTH = 600
+OFFSET = 150
+BLOCKSIZE = 50
+BLACK = (0, 0, 0)
+GREY = (160, 160, 160)
+WHITE = (200, 200, 200)
+GREEN = '#00A14B'
+RED = '#ED1C24'
+BLUE = '#21409A'
+YELLOW = '#FFDE17'
+PURPLE = '#7F3F98'
+CYAN = '#009599'
+PINK = (255, 153, 255)
+BROWN = (255, 128, 0)
+ORANGE = '#F26522'
+colors = [
+    GREEN, RED, BLUE, YELLOW, PURPLE, CYAN, ORANGE, BROWN,
+]
+
+
+def rand_color():
+    return random.choice(colors)
+
+
+def normalize_cords(x, y):
+    x = x - (x % 50)
+    y = y - (y % 50)
+    return (x, y)
+
+
+class AlignIt:
+    dim = 9
+
+    def __init__(self):
+        self.sqr_grid = [[0 for _ in range(self.dim)] for _ in range(self.dim)]
+        self.space = [[0 for _ in range(self.dim)] for _ in range(self.dim)]
+        self.next_sqrs = []
+        self.score = 0
+        self.selected_square = None
+        self.grow = True
+        self.move_made = True
+        self.same_color_counter = 0
+
+        self.main()
+
+    def setup_game(self, next_colors):
+        global SCREEN, CLOCK
+        pygame.init()
+        self.text_font = pygame.font.SysFont('Arial', 30)
+        pygame.mixer.music.pause()
+        SCREEN = pygame.display.set_mode((WINDOW_WIDTH, WINDOW_HEIGHT))
+        CLOCK = pygame.time.Clock()
+        SCREEN.fill(BLACK)
+        self.draw_future_grid(next_colors)
+        self.draw_grid(True)
+
+    def main(self):
+        next_colors = [rand_color() for _ in range(3)]
+        self.setup_game(next_colors)
+
+        while True:
+            self.draw_grid(False)
+            if self.move_made:
+                predicted_col_cords = self.draw_predicted(next_colors)
+                for x_grid, y_grid, color in predicted_col_cords:
+                    lines = self.find_adjacent_color((x_grid, y_grid), color)
+                    self.check_length_remove_square(lines)
+                    print(lines)
+                next_colors = [rand_color() for _ in range(3)]
+                self.draw_future_grid(next_colors)
+                self.move_made = False
+            self.handle_mouse_click()
+            self.handle_selected_square()
+            pygame.display.update()
```

Comment:
A good comment because `pygame.display.update()` does seem to get called in many redundant places - so there's definitely room for optimization.
But this comment is not "deterministically important" to become a test - it would be fine if we had looked over it.